packages/compadre/CMakeLists.txt

```diff
@@ -0,0 +1,617 @@
+  string(REPLACE "." ";" VLIST ${Compadre_VERSION})
+  list(GET VLIST 0 Compadre_VERSION_MAJOR)
+  list(GET VLIST 1 Compadre_VERSION_MINOR)
+  list(GET VLIST 2 Compadre_VERSION_PATCH)
+
+else() # Raw CMake Project
+
+  ##########
+  #
+  #   Compadre Details
+  #
+  ########## 
+
+  PROJECT(Compadre VERSION ${Compadre_VERSION} LANGUAGES CXX)
+
+  get_directory_property(Compadre_HAS_PARENT PARENT_DIRECTORY)
+
+  bob_begin_package()
+
+  # Set to OFF for significantly faster performance and ON for error tracking
+  bob_option(Compadre_DEBUG "Run Compadre Toolkit in DEBUG mode" ON)
+  bob_option(Compadre_EXTREME_DEBUG "Run Compadre Toolkit in EXTREME DEBUG mode" OFF)
+
+  # RPATH related settings
+  # https://gitlab.kitware.com/cmake/community/wikis/doc/cmake/RPATH-handling
+  SET(CMAKE_CXX_STANDARD 11)
+  SET(CMAKE_SKIP_BUILD_RPATH  FALSE)
+  SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+  SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+  SET(CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_LIBDIR} ${CMAKE_INSTALL_RPATH})
+
+  # RPATH should always include the folder it is called from
+  bob_option(PYTHON_CALLING_BUILD "Python setuptools calling build" OFF)
+  bob_option(Compadre_USE_PYTHON "Use PYTHON" OFF)
+  if (Compadre_USE_PYTHON)
+    if (APPLE)
+      SET(CMAKE_BUILD_RPATH ${CMAKE_BUILD_RPATH} "@loader_path/")
+      SET(CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_RPATH} "@loader_path/")
+      if (NOT(PYTHON_CALLING_BUILD))
+        # If python is calling the build, it will provide these flags
+        SET(CMAKE_PYTHON_SHARED_LINKER_FLAGS "-undefined dynamic_lookup -nodefaultlibs")
+      endif()
+    else()
+      SET(CMAKE_BUILD_RPATH ${CMAKE_BUILD_RPATH} "$ORIGIN/")
+      SET(CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_RPATH} "$ORIGIN/")
+      SET(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-z,origin")
+    endif()
+  endif()
+
+
+
+  ##########
+  #
+  #
+  #  TPLS
+  #
+  #
+  ##########
+
+
+
+  #TRILINOS
+  bob_input(Trilinos_PREFIX "" PATH "Path to Trilinos install")
+  if (Trilinos_PREFIX)
+    if(NOT IS_ABSOLUTE ${Trilinos_PREFIX})
+      set(Trilinos_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/${Trilinos_PREFIX}")
+    endif()
+    set(CMAKE_PREFIX_PATH ${Trilinos_PREFIX} ${CMAKE_PREFIX_PATH})
+  endif()
+
+  #KOKKOS
+  bob_input(KokkosCore_PREFIX "" PATH "Path to KokkosCore install")
+  if (KokkosCore_PREFIX)
+    if(NOT IS_ABSOLUTE ${KokkosCore_PREFIX})
+      set(KokkosCore_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/${KokkosCore_PREFIX}")
+    endif()
+    set(CMAKE_PREFIX_PATH ${KokkosCore_PREFIX} ${CMAKE_PREFIX_PATH})
+  endif()
+
+  #KOKKOS-KERNELS
+  bob_input(KokkosKernels_PREFIX "" PATH "Path to KokkosKernels install")
+  if (KokkosKernels_PREFIX)
+    if(NOT IS_ABSOLUTE ${KokkosKernels_PREFIX})
+      set(KokkosKernels_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/${KokkosKernels_PREFIX}")
+    endif()
+    set(CMAKE_PREFIX_PATH ${KokkosKernels_PREFIX} ${CMAKE_PREFIX_PATH})
+  endif()
+  set(KOKKOS_IN_TRILINOS OFF)

```

Comment:
@kuberry will compadre be distributed outside of Trilinos or only inside Trilinos?
If it is distributed only within Trilinos then this CMake logic needs to be cleaned-up.

Reply:
@lucbv Compadre will be distributed also outside Trilinos
https://github.com/SNLComputation/compadre